### PR TITLE
Taskqueue load test

### PR DESCRIPTION
**Pull request by jeanleonov**
Makes progress towards AppScale/appscale-testing#312.
This PR is dependency for https://github.com/AppScale/appscale-testing/pull/328

This PR adds load test for TaskQueue. Currently it just pushes load to the service and makes sure that TaskQueue behaves consistently. It worth to add performance comparison to master in the future.

[README.md](https://github.com/jeanleonov/appscale/blob/fadb33234af9ae4d25df0dd139d03eae4288c327/AppTaskQueue/test/load/README.md) for more details.

### CODE REVIEW

**Comment by jeanleonov** [comment]
https://ocd.appscale.com:8080/view/All/job/TaskQueue-Load-Test/40/console

File: AppTaskQueue/test/suites/run-load-test.sh [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+
+
+log ""
+log "=============================================="
+log "=== Preparing Virtualenv for the load test ==="
+log "=============================================="
+# aiohttp lib which is used in e2e test requires Python>=3.5.3
+# test scripts uses new syntax for formatting strings (3.6+)
+PYTHON=$(realpath --strip ./python-3.6.6-plus)
+${HELPERS_DIR}/ensure-python-version.sh 3.6.6 --create-link "${PYTHON}"
```
Comment (cdonati) [inline AppTaskQueue/test/suites/run-load-test.sh]: I don't think we should be installing a different version of Python on a user's machine automatically if they just want to run a load test.

I would be more comfortable just exiting with an error message here if their version is not compatible.

File: AppTaskQueue/test/helpers/helpers/datastore_v3_pb2.py [hunk 2]
```diff
@@ -0,0 +1,3873 @@
+# Generated by the protocol buffer compiler.  DO NOT EDIT!
```
Comment (cdonati) [inline AppTaskQueue/test/helpers/helpers/datastore_v3_pb2.py]: I don't think we should include machine-generated code in the repo if we can avoid it.

File: AppTaskQueue/test/suites/run-load-test.sh (hunk 1 above)
Comment (jeanleonov) [reply AppTaskQueue/test/suites/run-load-test.sh]: Agree, it was over-engineering.

**Comment by jeanleonov** [comment]
https://ocd.appscale.com:8080/view/All/job/TaskQueue-Load-Test/47/

**Comment by jeanleonov** [comment]
Codacy complains about:

 1. `dirs` variable is not used:
    `for root, dirs, files in os.walk(logs_dir):`
    I do think that it's more readable to use unused var in this case. Alternatively it would have to be named `_` or `root` and `files` would have to be extracted by `item[0]` and `item[2]`.
 2. Method could be a function (or static method).
    ```
     def check_performance_and_exit(self):
       # TODO, compare it to some baseline
       sys.exit(0)
     ```
     I wouldn't like to address this issue as it's just TODO stub which is going to be replaced with a regular method.
 3. Unused variable `bytes_noise`:
    `bytes_work_times, bytes_noise = task.body.split(b':', 1)`
    This var just clarifies what is the rest of `task.body`.

**Comment by jeanleonov** [comment]
https://ocd.appscale.com:8080/view/All/job/TaskQueue-Load-Test/59/console
```
2018-09-26 04:34:46: INFO === Test Cassandra implementation of Pull Queues ===
INFO TaskQueue activity stats:
 - 30220 tasks added
 - 19650 times tasks were leased
 - 8 tasks exhausted retries
 - 13160 tasks were completed successfully
 - Test execution time is 4793s

ERROR TaskQueue activity log is not consistent:
 - 408 tasks were leased before previous lease was expired
 - 19699 tasks left stalled
 - 2647 tasks were deleted more than once during the test
 - Number of added tasks (30220) is not equal to sum of deleted (13160) and exhausted (8)

Total summary:
    37121   total requests
    29638   failed requests 79.84%
       19   min response time
    63618   avg response time
   247740   max response time
       19.9 requests/s


2018-09-26 05:55:04: INFO === Test Postgres implementation of Pull Queues ===
INFO TaskQueue activity stats:
 - 63240 tasks added
 - 100992 times tasks were leased
 - 0 tasks exhausted retries
 - 63240 tasks were completed successfully
 - Test execution time is 547s
INFO TaskQueue activity log looks consistent

Total summary:
    65338   total requests
     2616   failed requests 4.00%
        7   min response time
     6230   avg response time
    45974   max response time
      114.0 requests/s
```

**Pull request merged**